challenges/10exercises.js

```diff
@@ -1,26 +1,181 @@
 //and use it to translate your Christmas cards from English into Swedish.
 
-//Write a function findLongestWord() that takes an array of words and returns the length of the longest one.
+var lexicon = {
+"merry":"god",
+"christmas":"jul",
+"and":"och",
+"happy":"gott",
+"new":"nytt",
+"year":"år"
+}
+
+function translate(text) {
+  return text.split(" ").reduce( function(accumulator, word) {
+    if (lexicon[word]) {
+      accumulator.push(lexicon[word])
+    }
+    return accumulator
+  }, []).join(" ")
+}
+
+//Other ways of doing it:
+// var textArray = text.split(" "); //['merry', 'christmas']
+// var translatedText = [];
+//
+// textArray.forEach( function(word) {
+//    if (lexicon[word]) {
+//      translatedText.push(lexicon[word])
+//    }
+// })
+//
+// return translatedText.join(" ");
+
+// for ( var i = 0; i < textArray.length; i++ ) {
+//  var currentWord = textArray[i];
+//  for ( var english in lexicon ) {
+//    if ( currentWord === english ) {
+//      translatedText.push(lexicon[english]);
+//    }
+//  }
+// }
+//when word in 'text' string equals to key in 'christmasCard' object, return value of that key
+
+
+
+//8. Write a function findLongestWord() that takes an array of words and returns the length of the longest one.
+
+function findLongestWord(text){
+  var textArray = text.split(" ");
+  var longestWord = 0;
+  for (var i=0; i < textArray.length; i++) {
+    if (textArray[i].length > longestWord) {
+      longestWord = textArray[i].length;
+    }
+  }
+return longestWord; //what if we want it to return the actual word?
+}
+
+
+//9. Write a function filterLongWords() that takes an array of words and an integer i and returns the array of words that are longer than i.
+
+  function filterLongWords(array, integer) {
+    var filterLongWords = function(array, integer){
+      var length = array.length;
+      var longestWords = [];
+      for (i = 0; i < length; i++) {
+        if (array[i].length > integer) {
+          longestWords[longestWords.length] = array[i];
+        }
+      }
+      return longestWords;
+    }
+}
+
 
-//Write a function filterLongWords() that takes an array of words and an integer i and returns the array of words that are longer than i.
+// 10. Write a function charFreq() that takes a string and builds a frequency listing of the characters contained in it. Represent the frequency listing as a Javascript object. Try it with something like charFreq("abbabcbdbabdbdbabababcbcbab").
 
-//Write a function charFreq() that takes a string and builds a frequency listing of the characters contained in it. Represent the frequency listing as a Javascript object. Try it with something like charFreq("abbabcbdbabdbdbabababcbcbab").
+ function charFreq(string){
+    return string.split('').reduce( function (accumulator, char){
+     if (!accumulator[char]) {
+       accumulator[char] = 1
+     } else {
+       accumulator[char] += 1
+     }
+     return accumulator
+   }, {})
+}
```

Comment:
I like this solution. There is an even more concise way: 
```
+function charFreq( str ) {
 +  let characterFrequency = {}
 +  str.split( "" ).forEach( letter => characterFrequency[ letter ] = ( characterFrequency[letter] || 0 ) + 1 )
 +  return characterFrequency
 +}
```
The above example using this: ( characterFrequency[letter] || 0 ) + 1 ), will add one to the key value if it exists, and if it does not exist, will return 0 and then add one to give the key its initial value.